### DISCUSSION TITLE
Syntax Higlighment - Why WinMerge search for keywords/directives inside of strings ?

### DISCUSSION THREAD

**Discussion by mlipok** [WinMerge/winmerge#610] [Q&A] [answered]
On [AutoIt Forum](https://www.autoitscript.com/forum/topic/205034-winmerge-github-stackoverflow-autoit-syntax-highlighting/?tab=comments#comment-1474626) there was find some bugs in highlightment engine.

![image](https://user-images.githubusercontent.com/11089482/107158531-10e8e080-698b-11eb-9f51-e097edd117d6.png)

Why WinMerge search for keywords/directives inside of strings ?

**Comment by mlipok** [comment]
Btw.
I think that this is due to the fact that this user use single quotes, instead double quotes.

![image](https://user-images.githubusercontent.com/11089482/107160521-046a8500-6997-11eb-8a20-f809cf83e1ce.png)

Do WinMerge use only double quotes to find stirngs ?

Wondering if it could be added catching also single quotes and not only double quotes, as a string statementes in to ".../Externals/crystaledit/editlib/parsers/autoit.cpp"

Maybe here:
```
      //  String constant "...."
      if (dwCookie & COOKIE_STRING)
        {
          if (pszChars[I] == '"')
            {
              dwCookie &= ~COOKIE_STRING;
              bRedefineBlock = true;
            }
          continue;
        }
```
or here:
```
      //  Normal text
      if (pszChars[I] == '"')
        {
          DEFINE_BLOCK (I, COLORINDEX_STRING);
          dwCookie |= COOKIE_STRING;
          continue;
        }
```


ps.
I'm only AutoIt scripter/coder not C or C++.

**Comment by sdottaka** [comment, marked as answer]
Fixed by commit a9b9110

Thank you for the report.

**Reply by mlipok** [reply]
It's always a pleasure to work with professionals.
